From a6daa3b6b37505dd1eedd839805c2f37952e5c0b Mon Sep 17 00:00:00 2001
From: Ewoud Kohl van Wijngaarden <ewoud@kohlvanwijngaarden.nl>
Date: Sat, 8 Dec 2018 13:42:23 +0100
Subject: [PATCH] Ignore hidden files

On Gentoo there's always a .keep_<package>-<slot> (e.g.,
.keep_cronbase-0) file inside the cron directory to ensure it's not
removed ith rmdir. Since usernames never start with a dot, we can safely
filter out all hidden files.
---
 lib/puppet/provider/cron/crontab.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/puppet/provider/cron/crontab.rb b/lib/puppet/provider/cron/crontab.rb
index fec5511..9f7363f 100644
--- a/lib/puppet/provider/cron/crontab.rb
+++ b/lib/puppet/provider/cron/crontab.rb
@@ -282,7 +282,7 @@ Puppet::Type.type(:cron).provide(:crontab, parent: Puppet::Provider::ParsedFile,
     return unless File.readable?(CRONTAB_DIR)
     Dir.foreach(CRONTAB_DIR) do |file|
       path = "#{CRONTAB_DIR}/#{file}"
-      yield(file) if File.file?(path) && File.writable?(path)
+      yield(file) if File.file?(path) && File.writable?(path) && !file.start_with?('.')
     end
   end
 
-- 
2.19.2

